python/social_bot/models/pioneer2dx_noplugin/model.sdf

```diff
@@ -154,6 +180,15 @@
         </material>
       </visual>
     </link>
+    <joint type="fixed" name="camera_joint">
```

Comment:
This change make simple_navigation fail too. As simple_navigation assumes 3 control variables. The third one is the camera rotation.

Reply:
Change back to revolute

Reply:
Considering the observation does not include state of camera joint, would rotating the camera confuse the agent? For example, 180 degree rotated camera gives an image meaning an opposite  left & right.
As a 120 degree camera is wide enough, it's easy for the agent to rotate itself by the 2 wheels to seek the goal into its view. Maybe we can just remove the 3rd control variable and set the joint to fixed.
Also, we have to do some extra work to fix the joint when inserting pioneer as an agent to other environment that does not need a rotating camera.

Reply:
It's okay to remove this control. But you need to change SimpleNavigation accordingly to make it still work.

Reply:
ok, I'll do that.